test/Mono.Linker.Tests.Cases/UnreachableBlock/ComplexConditionsOptimized.cs

```diff
@@ -37,13 +47,117 @@ public static void Test ()
 				case 2:
 					Reached ();
 					break;
-				default: throw new ApplicationException ();
+				default:
+					throw new ApplicationException ();
 				}
 			}
 
-			// https://github.com/dotnet/linker/issues/2888
-			// Should be removed
 			[Kept]
+			[ExpectedInstructionSequence (new[] {
+				"ldc.i4.2",
+				"stloc.0",
+				"ldloc.0",
+				"ldc.i4.1",
+				"sub",
+				"brtrue il_10",
+				"call System.Void Mono.Linker.Tests.Cases.UnreachableBlock.ComplexConditionsOptimized/TestSwitch::Reached()",
+				"ret",
+				"call System.Void Mono.Linker.Tests.Cases.UnreachableBlock.ComplexConditionsOptimized/TestSwitch::Reached()",
+				"br.s il_a",
+			})]
+			public static void TestOffset ()
```

Comment:
Would it make sense to also add tests to the `ComplexConditions` - basically the debug version of these tests - since the IL patterns will be somewhat different in that case.

Reply:
I did only the most straightforward pattern. There are more including pattern matching, nullable, etc that I point I think we need to invest in so will just leave that for next time.